pkg/registry/core/service/storage/rest_test.go

```diff
@@ -1421,6 +1421,94 @@ func TestServiceRegistryExternalService(t *testing.T) {
 	}
 }
 
+func TestAllocateLoadBalancerNodePortsFalse(t *testing.T) {
```

Comment:
I'm wondering if this and TestAllocateLoadBalancerNodePortsTrue could be the same unit test with a test table with 4 cases:
* allocate=true, gate on
* allocate false, gate on
* allocate=true, gate off
* allocate=fase, gate off

Reply:
So something like:

```go
func TestAllocateLoadBalancerNodePorts(t *testing.T) {
   testcases := []struct{
        name                          string
        svc                             *api.Service
        expectNodePorts        bool
        allocateNodePortGate bool
   }{
       ....
   }

   for _, testcase := range testcases {
       t.Run(testcase.name, func(t *testing.T) {
             ....
       })     
   }
}

```